### PR TITLE
Install `bitsandbytes` to run skipped bitsandbytes tests

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Follow up to #19233

Fabric GPU CI:
```
plugins/precision/test_bitsandbytes.py::test_bitsandbytes_layers[args0-expected0] PASSED [ 37%]
plugins/precision/test_bitsandbytes.py::test_bitsandbytes_layers[args1-expected1] PASSED [ 37%]
plugins/precision/test_bitsandbytes.py::test_bitsandbytes_layers_meta_device[args0-expected0] XFAIL [ 38%]
plugins/precision/test_bitsandbytes.py::test_bitsandbytes_layers_meta_device[args1-expected1] PASSED [ 38%]
```

cc @borda @carmocca @justusschock @awaelchli

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

**Comment by awaelchli** [comment]
@carmocca There is one special tests that won't run now:

```py
@pytest.mark.skipif(_BITSANDBYTES_AVAILABLE, reason="bitsandbytes needs to be unavailable")
def test_bitsandbytes_plugin(monkeypatch):
```

Maybe we could figure out a way to test it in a different way.

**Comment by Borda** [comment]
Shall we enforce this in CI so these tests are not accidentally skipped with dependency updates?

**Comment by awaelchli** [comment]
@Borda How could it be enforced?

File: requirements/fabric/test.txt [hunk 1]
```diff
@@ -7,3 +7,4 @@ pytest-random-order ==1.1.0
 click ==8.1.7
 tensorboardX >=2.2, <2.7.0  # min version is set by torch.onnx missing attribute
 torchmetrics >=0.7.0, <1.3.0 # needed for using fixed compare_version
+bitsandbytes ==0.41.0  # strict
```
Comment (Borda) [inline requirements/fabric/test.txt:10]: Would it make sense to have it in https://github.com/Lightning-AI/pytorch-lightning/blob/master/requirements/fabric/strategies.txt ?

Comment (awaelchli) [reply requirements/fabric/test.txt:10]: Why? what's the difference?

**Comment by carmocca** [comment]
> @carmocca There is one special tests that won't run now:
> 
> ```python
> @pytest.mark.skipif(_BITSANDBYTES_AVAILABLE, reason="bitsandbytes needs to be unavailable")
> def test_bitsandbytes_plugin(monkeypatch):
> ```
> 
> Maybe we could figure out a way to test it in a different way.

This test is meant to run on CPU since everything is mocked. On CPU, there's no reason to install bitsandbytes.

That would be my suggestion, install only bnb on the cuda jobs and then all test will run in at least one hardware setup

**Comment by Borda** [comment]
> On CPU, there's no reason to install bitsandbytes.

and would harm us in any way if we install it as it is part of: https://github.com/Lightning-AI/pytorch-lightning/blob/6bc27d54a004a0e79a6c17958d1332f8fe7f7369/.github/workflows/ci-tests-fabric.yml#L140

**Review by Borda — APPROVED** [review]

**Pull request merged**